model_compression_toolkit/core/keras/graph_substitutions/substitutions/virtual_activation_weights_composition.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2022 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import Tuple
+from tensorflow.keras.layers import Dense, DepthwiseConv2D, Conv2D, Conv2DTranspose
+from model_compression_toolkit.core import common
+from model_compression_toolkit.core.common import BaseNode, Graph
+from model_compression_toolkit.core.common.logger import Logger
+from model_compression_toolkit.core.common.graph.graph_matchers import NodeOperationMatcher, EdgeMatcher
+from model_compression_toolkit.core.common.graph.virtual_activation_weights_node import VirtualActivationWeightsNode
+
+
+class VirtualActivationWeightsComposition(common.BaseSubstitution):
+    def __init__(self):
+        """
+        Matches: Edge (Activation) --> (DepthwiseConv2D, Conv2D, Dense, Conv2DTranspose)
+        """
+
+        weights_node = NodeOperationMatcher(DepthwiseConv2D) | \
+                       NodeOperationMatcher(Conv2D) | \
+                       NodeOperationMatcher(Dense) | \
+                       NodeOperationMatcher(Conv2DTranspose)
+
+        act_node = weights_node.logic_not()
```

Comment:
It's not just activation layers

Reply:
I don't understand (let's take this offline).